lib/options-manager.js

```
@@ -644,9 +644,22 @@ async function recursiveBuildTsConfig(tsConfig, tsConfigPath) {
 	}
 
 	// If any of the following are missing, then we need to look up the base config as it could apply
-	// Use node resolution
 	const require = createRequire(tsConfigPath);
-	const basePath = require.resolve(tsConfig.extends);
+
+	let basePath;
+	try {
+		basePath = require.resolve(tsConfig.extends);
+	} catch (error) {
+		// Tsconfig resolution is odd, It allows behavior that is not exactly like node resolution
+		// therefore we attempt to smooth this out here with this hack
+		try {
+			basePath = require.resolve(path.join(tsConfig.extends, 'tsconfig.json'));
+		} catch {
+			// Throw the orginal resolution error to let the user know their extends block is invalid
+			throw error;
```

alternatively we could just swallow this error, and when that happens we would just generate the needed tsconfig for the file. That may be the better way to handle this.

---

I don't think we should swallow errors like that. It sounds like a pain to debug that it's working, but with the incorrect config.